src/Pages/login.tsx

```diff
@@ -1,23 +1,44 @@
-import { ChangeEvent, useState } from "react";
-import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import { ChangeEvent, FormEvent, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
 
 const Login = () => {
-    const [username, setUserName] = useState<string>('');
+    const [email, setEmail] = useState<string>('');
     const [password, setPassword] = useState<string>('');
+    const [incorrectRes, setIncorrectRes] = useState<boolean>(false);
 
     const navigate = useNavigate();
+    const location = useLocation();
 
+    const logIn = (e: FormEvent<HTMLFormElement>) => {
+        e.preventDefault();    
+        axios.post('http://restapi.adequateshop.com/api/authaccount/login', {
+            email: email,
+            password: password
+        })
+            .then(function (response) {
+                if (response.data.message === 'success') {
+                    navigate('/homepage', { state: { email: email, message: 'You have logged in succssfully' } });
+                } else {
+                    setIncorrectRes(true);
+                    navigate('/', { state: { message: response.data.message } });
+                }
+            })
```

Comment:
@ShiranthaKellum, this is okay but in our company, we usually create different async functions in a different file as util. Also, the method you used to handle errors has some problems; instead, you should wrap this code block with a try-catch statement and handle the errors from there.